analysis/filter/StructureModifier/src/main/kotlin/de/maibornwolff/codecharta/filter/structuremodifier/NodeRemover.kt

```diff
@@ -29,18 +29,33 @@ class NodeRemover(private val project: Project) {
         ).build(cleanAttributeDescriptors = true)
     }
 
-    private fun filterNodes(path: List<String>, node: MutableNode): MutableNode {
-        val filteredChildren = node.children.filter { it.name != path.firstOrNull() || path.size > 1 }
-        node.children = filteredChildren.map {
-            filterNodes(path.drop(1), it)
-        }.toMutableSet()
-        return node
-    }
+    private fun removeNodes(paths: List<List<String>>): List<MutableNode> {
+        val rootNode = project.rootNode.toMutableNode()
+        for (path in paths) {
+            var currentNode = rootNode
+            if (path[0] != "root") {
+                logger.warn("Path to node has to start with root: ${path.joinToString("/")}")
+                continue
+            }
+
+            val pathWithoutRoot = path.drop(1)
+            for ((index, pathSegment) in pathWithoutRoot.withIndex()) {
+                val isLastSegment = index == pathWithoutRoot.lastIndex
```

Comment:
I would suggest swapping the position of index and pathWithoutRoot.lastIndex, as index gets changed in every loop and the other variable doesn't, but this is only nitpicking